runtime/compiler/x/codegen/J9TreeEvaluator.cpp

```diff
@@ -6827,6 +6793,24 @@ static void genInitArrayHeader(
          }
       }
 
+   // Clear padding after size field in array header
+   if (TR::Compiler->om.compressObjectReferences())
+      { // Clear padding after size field in discontiguous header layout (for 0 length arrays)
+      generateMemImmInstruction(TR::InstOpCode::S4MemImm4,
+         node,
+         generateX86MemoryReference(objectReg, fej9->getOffsetOfDiscontiguousArraySizeField() + 4, cg),
+         0,
+         cg);
+      }
+   else
+      { // Clear padding after size field in contiguous header layout
+      generateMemImmInstruction(TR::InstOpCode::S4MemImm4,
+         node,
+         generateX86MemoryReference(objectReg, fej9->getOffsetOfDiscontiguousArraySizeField(), cg),
+         0,
+         cg);
+      }
```

Comment:
As I mentioned earlier, I think clearing the padding field might need to take into consideration situations where the padding field might not exist.